x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/ResultSetTestCase.java

```diff
@@ -2470,6 +2497,48 @@ private static void indexTestFieldsDoc(String docId, Object... values) throws IO
         });
     }
 
+    private void createTestDataForVersionType() throws IOException {
+        createIndexWithMapping("test");
+        updateMapping("test", builder -> {
+            builder.startObject("name").field("type", "keyword").endObject();
+            builder.startObject("version").field("type", "version").endObject();
+        });
+
+        int docId = 1;
+        // first two deterministic values
+        index("test", "" + (docId++), builder -> {
+            builder.array("version", "1.3.0");
+            builder.array("name", "version 1.3.0");
+        });
+        index("test", "" + (docId++), builder -> {
+            builder.array("version", "1.11.0");
+            builder.array("name", "version 1.11.0");
+        });
+
+        // some higher versions
+        for (int i = 0; i < randomInt(10); i++) {
```

Comment:
It was an attempt to randomize the test dataset as much as possible, adding valid and invalid version values outside of the range of the test queries so that the tests are still predictable. Eg. test that random values that do not match the test criteria are not returned by mistake.